src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java

```diff
@@ -0,0 +1,114 @@
+package pl.futurecollars.invoicing.db.file;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.utils.FilesService;
+import pl.futurecollars.invoicing.utils.IdService;
+import pl.futurecollars.invoicing.utils.JsonService;
+
+public class FileBasedDatabase implements Database {
+
+  private final FilesService filesService;
+  private final JsonService jsonService;
+  private final IdService idService;
+  private final Path databasePath;
+  private final Path idPath;
+
+  public FileBasedDatabase(Path databasePath, Path idPath, FilesService filesService, JsonService jsonService, IdService idService)
```

Comment:
Baza danych raczej nie powinna wiedzieć o idPath. To powinno interesować jedynie IdService.